### ISSUE TITLE
Executing the previous run from command line

### ISSUE THREAD

**Issue by BugDiver** [getgauge/gauge#862]
User should be able to run the previous run from the command line.

**Proposed behavior**
* Add a flag `--repeat` to `run` subcommand.
* Execute the last run with same flags and env by running `gauge run --repeat` command.

**Comment by haroon-sheikh** [comment]
@BugDiver @riju91  What is the use case of this? 

It sounds like a good feature only if we could pass the number of times to repeat. This can be really useful when trying to replicate a intermitted issue.

E.g. `gauge run --repeat=3`. 

I guess the only problem will be reporting when running multiple times.

**Comment by sriv** [comment]
@haroon-sheikh - this issue is a pre-requisite for https://github.com/getgauge/gauge-vscode/issues/41

TLDR: we are implementing a feature in vscode plugin that will allow users to author implementation, and repeat scenario run without having to toggle between files.

**Comment by BugDiver** [comment]
should be available in nightly >= 20-11-2017

**Comment by sguptatw** [comment]
# Test cases
- [ ] Should be able to repeat run for passed, failed, skipped tests
- [ ] Should give an error message if there is no last run
- [ ] Should be able to repeat run with tags, parallel run, data driven execution, selective data driven execution, folder execution

**Comment by sguptatw** [comment]
If gauge does not execute specs
example: - 
- [ ] No specifications found in specs. - for a tagged expression
- [ ] Failed to load env. something1 environment does not exist - for a non existing env input

What should be the behavior of re-run in such scenarios

**Comment by zabil** [comment]
> What should be the behavior of re-run in such scenarios

It should show and info on the problems mentioned.

**Comment by sguptatw** [comment]
**Expected behavior**
Should show info when previous run has failed.

**Actual behavior**
Runs the previously successful gauge execution.

**Comment by BugDiver** [comment]
Should be fixed in nightly >= 27-11-2017

**Comment by sguptatw** [comment]
Please find the failing test cases inline in the previous comment

**Comment by sguptatw** [comment]
If the previous command is `gauge run --failed` what should `gauge run --repeat` do?

**Comment by sriv** [comment]
> If the previous command is gauge run --failed what should gauge run --repeat do when run multiple times? Should it run only the failed tests repeatedly?

IMO, `gauge run --repeat` should _just_ repeat the last run. It should be synonymous with running the previous command again.

In this case this:

```
> gauge run --failed
> gauge run --repeat
> gauge run --repeat
```
should be equivalent to:

```
gauge run --failed
gauge run --failed
gauge run --failed
```

This would imply that any test that passes in the 2nd run *will not* be repeated in the 3rd.

**Comment by riju91** [comment]
Should be available in nightly >= 12-12-2017
